File: InvotraxApp/frontend/invotrax/src/app/models/update-product-command.ts
import { BarcodeDto } from "./barcode.dto";
import { SerialNumberDto } from "./serial-number.dto";

export interface UpdateProductCommand {
    name: string;
    sku: string;
    description: string;
    productTypeId: number;
    categoryId: number;
    manufacturerId: number;
    netSellingPrice: number;
    grossSellingPrice: number;
    warrantyPeriodMonths: number;
    serialNumberRequired: boolean;
    vatRateId: number;
    unit: string;
    deleted: boolean;
    barcodes: BarcodeDto[];
    serialNumbers: SerialNumberDto[];

}

File: InvotraxApp/backend/InvotraxApp/src/main/java/dev/lkeleti/invotraxapp/dto/UpdateProductCommand.java
package dev.lkeleti.invotraxapp.dto;

import dev.lkeleti.invotraxapp.model.Barcode;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class UpdateProductCommand {
    private String name;
    private String sku;
    private String description;
    private String unit;
    private Long categoryId;
    private Long manufacturerId;
    private BigDecimal netPurchasePrice;
    private BigDecimal grossPurchasePrice;
    private BigDecimal netSellingPrice;
    private BigDecimal grossSellingPrice;
    private Integer warrantyPeriodMonths;
    private boolean serialNumberRequired;
    private int stockQuantity;
    private Long vatRateId;
    private long productTypeId;
    private boolean deleted;
    private List<BarcodeDto> barcodes;
    private List<SerialNumberDto> serialNumbers;
}
